### DISCUSSION TITLE
OSCP Approved?

### DISCUSSION THREAD

**Discussion by MacGeiver50** [calebstewart/pwncat#162] [Q&A] [answered]
I have been working through my PWK course and was wondering if pwncat would flag as an illegal tool for the OSCP exam? Please advise as I would like to start incorporating the tool now if pwncat is allowed.

**Comment by calebstewart** [comment, marked as answer]
I think that's a better question for Offensive Security themselves, to be honest. Their exam guidelines regarding tool usage are intentionally vague, as they aren't capable of evaluating every tool in existence. In general, you shouldn't use things that "automate" actions for you. pwncat is capable of doing that, though I can't speak for Offensive Security.

If I were taking the OSCP now, I would probably ere on the side of caution and not use it. If you reach out to Offensive Security, and they give you a straight answer (which is unlikely), then please post back here and let me/anyone else looking for a straightforward answer know.

**Reply by MacGeiver50** [reply]
Thank you for the detailed response! I will reach out tomorrow and reply back with what I find out from Offensive Security.

**Reply by MacGeiver50** [reply]
Good day,
I did reach out to Offensive Security today and they replied back very quickly with the below response. Looks like I will need to dig a little deeper info the functionality of pwncat and what services that it offers that will not compromise the auto exploitation of a shell etc. Thanks again for all the help with my noob questions. 


Please note that tools that perform automatic enumeration are allowed during the exam, however, tools that perform automatic exploitation are not allowed during the exam.

In the OSCP Exam Guide, we have provided a number of categories of tools that are currently restricted.
Students are responsible for reviewing and knowing the features used by their chosen tools or scripts and should determine if they are in any restricted categories.
As long as the tool or script does not use any prohibited features or falls under any of the restricted categories mentioned in the exam guide, then you are free to use it.

For more information about the exam restrictions please visit: https://help.offensive-security.com/hc/en-us/articles/360040165632-OSCP-Exam-Guide#exam-restrictions
For more information on understanding the tools/scripts that you use in a pentest, please visit: https://www.offensive-security.com/offsec/understanding-pentest-tools-scripts/
